app/forms/sign_in_or_continue_form.rb

```diff
@@ -0,0 +1,43 @@
+class SignInOrContinueForm < Form
+  # - Sign in with Teacher ID does NOT save or interact with this Form and POSTs straight to Teacher ID
+  # - If Teacher ID is disabled for the journey, we still mimick if a user clicked on "Continue without signing in".
+
+  def initialize(claim:, journey:, params:)
+    super
+
+    skip_form_if_teacher_id_not_enabled_for_journey
+  end
+
+  def save
+    continue_without_teacher_id
+
+    true
+  end
+
+  # This is needed because logging in with Teacher ID doesn't submit to this form.
+  # Therefore the slug/page needs to be considered submitted to avoid it being the next slug
+  # after callback redirect back to Claim from Teacher ID.
```

Comment:
Untested theory, but I wonder if this form should be combined with @rjlynch 's work - see my comment here https://github.com/DFE-Digital/claim-additional-payments-for-teaching/pull/2690#discussion_r1570647909

I think by incorporating the callback functionality into this form object we might be able to avoid these workarounds.

Reply:
I'm unsure about combining the forms at the minute but I think we could move the logic for updating the slug to [here](https://github.com/DFE-Digital/claim-additional-payments-for-teaching/blob/9fdbc64600fc6f8cd1cf2f52fa347c34006c99bb/app/controllers/omniauth_callbacks_controller.rb#L14) in the call back controller to avoid this form having to care about it.
That controller already knows about the pages in the journey (it redirects to a specific page) so while not ideal I don't think it does much harm modifying the session slugs there.